### PR TITLE
METRON-2251: Update with the latest master branch

### PR DESCRIPTION
**Pull request by sardell**
## Contributor Comments
This change merges the latest from master into the `feature/METRON-1856-parser-aggregation` feature branch. 

Because of the age of this branch, this merge with master resulted in some merge conflicts. In addition, both UIs have major version updates to dependencies like Angular and Typescript. As a result, I had to update a few areas of the parser aggregation code to accommodate for stricter type checking.

This PR will show all of the commits from master that are being merged into the feature branch, followed by [a merge commit](https://github.com/apache/metron/commit/24f3b7198a18a89763d64b6b535f51a5235b7e78). After the merge commit, there are 2 commits. One is to apply an object property to a class that had been refactored in our parser aggregation work, and the other commit is to fix type errors resulting from stricter type checking.

## Testing
The easiest way to do this is by building full dev from scratch, but another faster way is to navigate to the Management UI directory under `metron-interface/metron-config`, run the unit tests by running `ng test` and run  `ng build --prod` to ensure that the build passes.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [ ] n/a ~Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:~

  ```
  cd site-book
  mvn site
  ```

- [ ] n/a ~Have you ensured that any documentation diagrams have been updated, along with their source files, using [draw.io](https://www.draw.io/)? See [Metron Development Guidelines](https://cwiki.apache.org/confluence/display/METRON/Development+Guidelines) for instructions.~

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-interface/metron-config/src/app/sensors/effects/sensors.effects.spec.ts [hunk 1]
```diff
@@ -129,16 +129,18 @@ describe('sensor.effects.ts', () => {
     store.dispatch(new fromActions.ApplyChanges());
   });
 
-  it('Should show notification when operation SUCCESSFULL', () => {
-    store.dispatch(new fromActions.ApplyChanges());
-    expect(userNotificationSvc.showSuccessMessage).toHaveBeenCalled();
-  });
+  // Removing these tests until we rewrite this test suite to use a mock store and mock actions
```
Comment (mmiklavc) [inline metron-interface/metron-config/src/app/sensors/effects/sensors.effects.spec.ts]: Does this have tasks added under the Jira for feature branch completion? Please link to this PR.

**Comment by mmiklavc** [comment]
I looked at the diff as follows - https://github.com/apache/metron/pull/1507/files/24f3b7198a18a89763d64b6b535f51a5235b7e78..908a26f5da80563acb271424d28736214867dfff

This isolated 4 commits with pertinent changes made in this merge PR.

@sardell The only question I have is regarding the commented out tests.

File: metron-interface/metron-config/src/app/sensors/effects/sensors.effects.spec.ts (hunk 1 above)
Comment (sardell) [inline metron-interface/metron-config/src/app/sensors/effects/sensors.effects.spec.ts]: JIRA related to this commented code: https://issues.apache.org/jira/browse/METRON-2277

Comment (sardell) [reply metron-interface/metron-config/src/app/sensors/effects/sensors.effects.spec.ts]: It does now: https://issues.apache.org/jira/browse/METRON-2277

Thanks for pointing this out, @mmiklavc.

**Comment by mmiklavc** [comment]
Ok, go for it +1. Thanks @sardell

**Pull request closed without merging**